### PR TITLE
Adds Voidwalker to the event controller

### PR DESCRIPTION
**Pull request by StrangeWeirdKitten**
## About The Pull Request

Adds an entry into the event controller for Void Walker. I just used the dynamic voidwalker weights and times.

## Why It's Good For The Game

Every other midround off the top of my head is located inside the event controller. Dynamic is an entirely seperate system, yes. It would be nice for it to be able to be inside the event panel aswell. 

## Changelog

:cl:
code: Void Walker has been added to the event controller and given a signup icon.
/:cl:

### CODE REVIEW

File: code/modules/events/ghost_role/voidwalker.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	if(isnull(chosen_one))
+		return NOT_ENOUGH_PLAYERS
+
+	var/datum/mind/player_mind = new /datum/mind(chosen_one.key)
+	player_mind.active = TRUE
+	var/mob/living/carbon/human/walker = new (spawn_location)
+	player_mind.transfer_to(walker)
+	player_mind.set_assigned_role(SSjob.GetJobType(/datum/job/voidwalker))
+	player_mind.add_antag_datum(/datum/antagonist/voidwalker)
+	walker.set_species(/datum/species/voidwalker)
```
Comment (Time-Green) [inline code/modules/events/ghost_role/voidwalker.dm:40]:
```suggestion
```
This is already set by the antag datum

File: code/modules/events/ghost_role/voidwalker.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+/datum/round_event/ghost_role/void_walker
+	minimum_required = 40
+	fakeable = FALSE
+	role_name = "void walker"
+
+/datum/round_event/ghost_role/void_walker/spawn_role()
+	var/spawn_location = find_space_spawn()
+	if(isnull(spawn_location))
+		return MAP_ERROR
```
Comment (Time-Green) [inline code/modules/events/ghost_role/voidwalker.dm:28]: Isn't this already checked by can_spawn_event?

File: code/modules/events/ghost_role/voidwalker.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/datum/round_event_control/voidwalker
+	name = "Spawn Void Walker"
```
Comment (Time-Green) [inline code/modules/events/ghost_role/voidwalker.dm:2]:
```suggestion
	name = "Spawn Voidwalker"
```

File: code/modules/events/ghost_role/voidwalker.dm [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+	typepath = /datum/round_event/ghost_role/void_walker
+	max_occurrences = 1
+	weight = 2
+	min_players = 40
+	dynamic_should_hijack = TRUE
+	category = EVENT_CATEGORY_ENTITIES
+	description = "A Void Walker that rips people out of the station and into the abyss"
```
Comment (Time-Green) [inline code/modules/events/ghost_role/voidwalker.dm:9]:
```suggestion
	description = "A Voidwalker that rips people out of the station and into the abyss"
```

File: code/modules/events/ghost_role/voidwalker.dm [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+/datum/round_event_control/ghost_role/voidwalker/can_spawn_event(players_amt, allow_magic = FALSE)
+	var/space_turf = find_space_spawn()
+	if(!space_turf)
+		return FALSE
+	return ..()
+
+/datum/round_event/ghost_role/void_walker
+	minimum_required = 40
+	fakeable = FALSE
+	role_name = "void walker"
```
Comment (Time-Green) [inline code/modules/events/ghost_role/voidwalker.dm:23]:
```suggestion
	role_name = "voidwalker"
```

File: code/modules/events/ghost_role/voidwalker.dm [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+/datum/round_event/ghost_role/void_walker/spawn_role()
+	var/spawn_location = find_space_spawn()
+	if(isnull(spawn_location))
+		return MAP_ERROR
+
+	var/mob/chosen_one = SSpolling.poll_ghost_candidates(check_jobban = ROLE_VOIDWALKER, role = ROLE_VOIDWALKER, alert_pic = /obj/item/cosmic_skull, jump_target = spawn_location, role_name_text = "Void Walker", amount_to_pick = 1)
```
Comment (Time-Green) [inline code/modules/events/ghost_role/voidwalker.dm:30]:
```suggestion
	var/mob/chosen_one = SSpolling.poll_ghost_candidates(check_jobban = ROLE_VOIDWALKER, role = ROLE_VOIDWALKER, alert_pic = /obj/item/cosmic_skull, jump_target = spawn_location, role_name_text = "Voidwalker", amount_to_pick = 1)
```

**Review by Time-Green — CHANGES_REQUESTED** [review]
What are the consequences of this? Doesn't this mean voidwalkers spawn way more since they're now being spawned by two different systems?

**Comment by Jacquerel** [comment]
I do not think we need every midround to be in the event controller. As far as I am aware this was only true of older antagonists for legacy reasons.
You need to provide an independent justification for this change which is not "other things do it".

**Comment by StrangeWeirdKitten** [comment]
> What are the consequences of this? Doesn't this mean voidwalkers spawn way more since they're now being spawned by two different systems? 

Then I'll go ahead and set the max_occurrence to 0. I ain't trying to add it for it rolling. All I'm trying to do is add it to trigger-event verb. Little nitpicky I suppose? Yeah sure, lol.

**Comment by StrangeWeirdKitten** [comment]
Talked with Time-Green about it.

**Pull request closed without merging**